gensim/models/tfidfmodel.py

```diff
@@ -152,10 +159,37 @@ def __getitem__(self, bow, eps=1e-12):
         # and finally, normalize the vector either to unit length, or use a
         # user-defined normalization function
         if self.normalize is True:
-            vector = matutils.unitvec(vector)
+            norm_vector = matutils.unitvec(vector)
         elif self.normalize:
-            vector = self.normalize(vector)
+            norm_vector = self.normalize(vector)
 
         # make sure there are no explicit zeroes in the vector (must be sparse)
-        vector = [(termid, weight) for termid, weight in vector if abs(weight) > eps]
+        sparse_norm_vector = [
+            (termid, weight) for termid, weight in norm_vector if                       abs(weight) > eps
+        ]
+
+        n_samples = len(self.idfs)
+        piv_lis = [0]*n_samples
```

Comment:
PEP8: spaces around operator (here and everywhere)

Reply:
Also, maybe create it as `piv_lis = np.zeros(n_samples, dtype=int)` (you convert it to numpy-array later), same for `lis`